Build failure: obs-studio-plugins.advanced-scene-switcher

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

1. Build *obs-studio-plugins.advanced-scene-switcher*

### Can Hydra reproduce this build failure?

Please select the Hydra Status.

### Link to Hydra build job

N/A

### Relevant log output

```console
this derivation will be built:
  /nix/store/7qj3m8s08jibwx41y2khg1zm3czz1f6g-advanced-scene-switcher-1.28.1.drv
building '/nix/store/7qj3m8s08jibwx41y2khg1zm3czz1f6g-advanced-scene-switcher-1.28.1.drv'...
Running phase: qtPreHook
Running phase: unpackPhase
unpacking source archive /nix/store/rf3r4n2gdkb1z3axqh5wj3bv4fg2pr3v-source
source root is source
Running phase: patchPhase
applying patch /nix/store/c51a4qkji554vyzbldadnhy0lq9yf4qz-e0c650574f9f7f6cae5626afa9abf8a838dc0858.diff
patching file .cmake-format.json
patching file .github/workflows/build-debian.yml
patching file CMakeLists.txt
Hunk #1 succeeded at 289 (offset 2 lines).
patching file build-aux/CI/linux/debian/control
patching file cmake/common/advss_helpers.cmake
patching file cmake/common/helpers_common.cmake
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
fixing cmake files...
cmake flags: -GNinja -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/3m224nj8lprc5zcg4zzklc7g5a5lscxz-advanced-scene-switcher-1.28.1/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/3m224nj8lprc5zcg4zzklc7g5a5lscxz-advanced-scene-switcher-1.28.1/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/3m224nj8lprc5zcg4zzklc7g5a5lscxz-advanced-scene-switcher-1.28.1/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/3m224nj8lprc5zcg4zzklc7g5a5lscxz-advanced-scene-switcher-1.28.1/share/doc/advanced-scene-switcher -DCMAKE_INSTALL_INFODIR=/nix/store/3m224nj8lprc5zcg4zzklc7g5a5lscxz-advanced-scene-switcher-1.28.1/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/3m224nj8lprc5zcg4zzklc7g5a5lscxz-advanced-scene-switcher-1.28.1/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/3m224nj8lprc5zcg4zzklc7g5a5lscxz-advanced-scene-switcher-1.28.1/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/3m224nj8lprc5zcg4zzklc7g5a5lscxz-advanced-scene-switcher-1.28.1/include -DCMAKE_INSTALL_SBINDIR=/nix/store/3m224nj8lprc5zcg4zzklc7g5a5lscxz-advanced-scene-switcher-1.28.1/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/3m224nj8lprc5zcg4zzklc7g5a5lscxz-advanced-scene-switcher-1.28.1/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/3m224nj8lprc5zcg4zzklc7g5a5lscxz-advanced-scene-switcher-1.28.1/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/591apldfgm1gr238rbfc8ib0y1z8ng2y-gcc-wrapper-14-20241116/bin/strip -DCMAKE_RANLIB=/nix/store/591apldfgm1gr238rbfc8ib0y1z8ng2y-gcc-wrapper-14-20241116/bin/ranlib -DCMAKE_AR=/nix/store/591apldfgm1gr238rbfc8ib0y1z8ng2y-gcc-wrapper-14-20241116/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/3m224nj8lprc5zcg4zzklc7g5a5lscxz-advanced-scene-switcher-1.28.1
-- The C compiler identification is GNU 14.2.1
-- The CXX compiler identification is GNU 14.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/591apldfgm1gr238rbfc8ib0y1z8ng2y-gcc-wrapper-14-20241116/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/591apldfgm1gr238rbfc8ib0y1z8ng2y-gcc-wrapper-14-20241116/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- CMAKE_PROJECT_NAME is advanced-scene-switcher
-- advanced-scene-switcher configured for out-of-tree build
-- Found Threads: TRUE
-- Performing Test HAVE_STDATOMIC
-- Performing Test HAVE_STDATOMIC - Success
-- Found WrapAtomic: TRUE
-- Found OpenGL: /nix/store/rvlpzcdms824wjg8q31bdfvx35z6zf27-libglvnd-1.7.0/lib/libOpenGL.so
-- Found WrapOpenGL: TRUE
-- Found XKB: /nix/store/s444n5f48zm5avq8szkzzfwz2s4mjl9d-libxkbcommon-1.7.0/lib/libxkbcommon.so (found suitable version "1.7.0", minimum required is "0.5.0")
-- Found WrapVulkanHeaders: /nix/store/agb2xa0zs3qx7r5xxkx85yd273aqm0rz-vulkan-headers-1.3.296.0/include
-- Found nlohmann_json: /nix/store/my7lz8jpdmq609rwgi0xvkzcy9148s0m-nlohmann_json-3.11.3/share/cmake/nlohmann_json/nlohmann_jsonConfig.cmake (found version "3.11.3")
-- Found X11: /nix/store/b0w4jcxry1hc1yqfg7g6289jc3j7fwnq-xorgproto-2024.1/include  found components: Xss
-- Looking for XOpenDisplay in /nix/store/36ybrb8mfqbvq0irggdqhd85iqabfmk9-libX11-1.8.10/lib/libX11.so;/nix/store/ffgbdyv2lwg1ckfqy4hck7sf1s754k5c-libXext-1.3.6/lib/libXext.so
-- Looking for XOpenDisplay in /nix/store/36ybrb8mfqbvq0irggdqhd85iqabfmk9-libX11-1.8.10/lib/libX11.so;/nix/store/ffgbdyv2lwg1ckfqy4hck7sf1s754k5c-libXext-1.3.6/lib/libXext.so - found
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for connect
-- Looking for connect - found
-- Looking for remove
-- Looking for remove - found
-- Looking for shmat
-- Looking for shmat - found
-- advanced-scene-switcher using libproc2
-- Found X11: /nix/store/b0w4jcxry1hc1yqfg7g6289jc3j7fwnq-xorgproto-2024.1/include  found components: Xtst
-- ADVSS: ENABLED PLUGIN     advanced-scene-switcher-base
-- Performing Test LIBREMIDI_HAS_STD_JTHREAD
-- Performing Test LIBREMIDI_HAS_STD_JTHREAD - Failed
-- Performing Test LIBREMIDI_HAS_STD_SEMAPHORE
-- Performing Test LIBREMIDI_HAS_STD_SEMAPHORE - Failed
CMake Error at /nix/store/mfrfb675a16gqchycqv16aziivn09dxa-cmake-3.31.3/share/cmake-3.31/Modules/ExternalProject/shared_internal_commands.cmake:943 (message):
  error: could not find git for clone of readerwriterqueue-populate
Call Stack (most recent call first):
  /nix/store/mfrfb675a16gqchycqv16aziivn09dxa-cmake-3.31.3/share/cmake-3.31/Modules/ExternalProject.cmake:3041 (_ep_add_download_command)
  CMakeLists.txt:29 (ExternalProject_Add)


-- Configuring incomplete, errors occurred!

CMake Error at /nix/store/mfrfb675a16gqchycqv16aziivn09dxa-cmake-3.31.3/share/cmake-3.31/Modules/FetchContent.cmake:1906 (message):
  CMake step for readerwriterqueue failed: 1
Call Stack (most recent call first):
  /nix/store/mfrfb675a16gqchycqv16aziivn09dxa-cmake-3.31.3/share/cmake-3.31/Modules/FetchContent.cmake:1609 (__FetchContent_populateSubbuild)
  /nix/store/mfrfb675a16gqchycqv16aziivn09dxa-cmake-3.31.3/share/cmake-3.31/Modules/FetchContent.cmake:2145:EVAL:2 (__FetchContent_doPopulation)
  /nix/store/mfrfb675a16gqchycqv16aziivn09dxa-cmake-3.31.3/share/cmake-3.31/Modules/FetchContent.cmake:2145 (cmake_language)
  /nix/store/mfrfb675a16gqchycqv16aziivn09dxa-cmake-3.31.3/share/cmake-3.31/Modules/FetchContent.cmake:2384 (__FetchContent_Populate)
  deps/libremidi/cmake/libremidi.deps.cmake:37 (FetchContent_MakeAvailable)
  deps/libremidi/CMakeLists.txt:50 (include)


-- Configuring incomplete, errors occurred!
error: builder for '/nix/store/7qj3m8s08jibwx41y2khg1zm3czz1f6g-advanced-scene-switcher-1.28.1.drv' failed with exit code 1;
       last 25 log lines:
       > -- Performing Test LIBREMIDI_HAS_STD_JTHREAD
       > -- Performing Test LIBREMIDI_HAS_STD_JTHREAD - Failed
       > -- Performing Test LIBREMIDI_HAS_STD_SEMAPHORE
       > -- Performing Test LIBREMIDI_HAS_STD_SEMAPHORE - Failed
       > CMake Error at /nix/store/mfrfb675a16gqchycqv16aziivn09dxa-cmake-3.31.3/share/cmake-3.31/Modules/ExternalProject/shared_internal_commands.cmake:943 (message):
       >   error: could not find git for clone of readerwriterqueue-populate
       > Call Stack (most recent call first):
       >   /nix/store/mfrfb675a16gqchycqv16aziivn09dxa-cmake-3.31.3/share/cmake-3.31/Modules/ExternalProject.cmake:3041 (_ep_add_download_command)
       >   CMakeLists.txt:29 (ExternalProject_Add)
       >
       >
       > -- Configuring incomplete, errors occurred!
       > 
       > CMake Error at /nix/store/mfrfb675a16gqchycqv16aziivn09dxa-cmake-3.31.3/share/cmake-3.31/Modules/FetchContent.cmake:1906 (message):
       >   CMake step for readerwriterqueue failed: 1
       > Call Stack (most recent call first):
       >   /nix/store/mfrfb675a16gqchycqv16aziivn09dxa-cmake-3.31.3/share/cmake-3.31/Modules/FetchContent.cmake:1609 (__FetchContent_populateSubbuild)
       >   /nix/store/mfrfb675a16gqchycqv16aziivn09dxa-cmake-3.31.3/share/cmake-3.31/Modules/FetchContent.cmake:2145:EVAL:2 (__FetchContent_doPopulation)
       >   /nix/store/mfrfb675a16gqchycqv16aziivn09dxa-cmake-3.31.3/share/cmake-3.31/Modules/FetchContent.cmake:2145 (cmake_language)
       >   /nix/store/mfrfb675a16gqchycqv16aziivn09dxa-cmake-3.31.3/share/cmake-3.31/Modules/FetchContent.cmake:2384 (__FetchContent_Populate)
       >   deps/libremidi/cmake/libremidi.deps.cmake:37 (FetchContent_MakeAvailable)
       >   deps/libremidi/CMakeLists.txt:50 (include)
       >
       > 
       > -- Configuring incomplete, errors occurred!
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.9-xanmod1, NixOS, 25.05 (Warbler), 25.05beta742084.9e4d5190a948`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - channels(root): `"home-manager, nixos, nur"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

### Notify maintainers


@PatrickDaG

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/video/obs-studio/plugins/advanced-scene-switcher/default.nix
+++ b/pkgs/applications/video/obs-studio/plugins/advanced-scene-switcher/default.nix
@@ -88,7 +88,7 @@ stdenv.mkDerivation rec {
 
   # PipeWire support currently disabled in libremidi dependency.
   # see https://github.com/NixOS/nixpkgs/pull/374469
-  cmakeFlags = [ "-DLIBREMIDI_NO_PIPEWIRE=ON" ];
+  cmakeFlags = [ (lib.cmakeBool "LIBREMIDI_NO_PIPEWIRE" true) ];
 
   env.NIX_CFLAGS_COMPILE = "-Wno-error=stringop-overflow -Wno-error=deprecated-declarations";
 

--- a/pkgs/applications/video/obs-studio/plugins/advanced-scene-switcher/default.nix
+++ b/pkgs/applications/video/obs-studio/plugins/advanced-scene-switcher/default.nix
@@ -86,6 +86,10 @@ stdenv.mkDerivation rec {
     chmod -R +w $sourceRoot/deps/libremidi
   '';
 
+  # PipeWire support currently disabled in libremidi dependency.
+  # see https://github.com/NixOS/nixpkgs/pull/374469
+  cmakeFlags = [ "-DLIBREMIDI_NO_PIPEWIRE=ON" ];
+
   env.NIX_CFLAGS_COMPILE = "-Wno-error=stringop-overflow -Wno-error=deprecated-declarations";
 
   passthru.updateScript = nix-update-script { };

